docs/sorting-algorithms/heap-sort.mdx

```diff
@@ -0,0 +1,326 @@
+---
+title: Heap Sort
+slug: /sorting-algorithms/heap-sort
+---
+
+# Introduction
+
+Heap Sort is a sorting algorithm which extracts the maximum or minimum element from the array and swap it with the last element of the array for sorting in ascending or descending order N times.
+It is similar to Selection Sort, the only difference is that it uses Heap data structure for finding minimum or maximum element.
+
+# Algorithm
+
+In Heap Sort, we follow the following algorithm:
+
+* First we will convert array into MaxHeap
+
+* Swap the root of the MaxHeap with the last element of the MaxHeap 
+
+* Reduce the size of MaxHeap by 1 from the last and call Heapify on the root.
+
+* Perform step 2 and step 3 N-1 times
+
+# Explanation for Heap Sort
+
+### Example 
+
+```
+ | 4 | 1 | 2 | 7 |
+
+ Converting it to MaxHeap:
+ | 7 | 4 | 2 | 1 |
+
+ Swaping first and last element of Heap and reducing the size by 1
+ | 1 | 4 | 2 || 7 |
+
+ Calling Heapify on root
+ | 4 | 1 | 2 || 7 |
+
+ Swaping first and last element of Heap and reducing the size by 1
+ | 2 | 1 || 4 | 7 |
+
+ Calling Heapify on root
+ | 2 | 1 || 4 | 7 |
+
+ Swaping first and last element of Heap and reducing the size by 1
+ | 1 || 2 | 4 | 7 |
+```
```

Comment:
Can you use a more intuitive example, please?

Reply:
done